### ISSUE TITLE
Search is returning "no results" for words that are in Pages and Posts

### ISSUE THREAD

**Issue by michaellenaghan** [typlog/issues#11]

**Comment by michaellenaghan** [comment]
It seems like basic search only searches titles and subtitles.

**Comment by lepture** [comment]
Yes. Basic search only supports titles and subtitles. You can turn on Algolia integration to support full text search.

**Comment by michaellenaghan** [comment]
Just to clarify: I've been trying to create issues for things that surprised me as I was getting to know Typlog. I'm hoping that that "first impression" feedback is helpful.

Of course, there are different kinds of surprises. Today some of the surprises happened because Typlog didn't follow existing conventions. That created confusion. Not just that: it created the impression that there were bugs.

The More link at the bottom of the Home page was one example. The convention is that posts are paged, and that the last page of posts doesn't have a More link. Typlog doesn't actually page posts. That's not necessarily bad. But then Typlog latches on to an existing convention, More, and uses it in a different way. I think that if you're going to break convention you have to _signal_ that you're breaking convention. (I think you should also ask: is it worth the cost of breaking this convention?) For example, that's what was behind my suggestion that you label the link "Archive" rather than "More". (I think there's still a problem with that solution, but I'll save that for the correct issue; still, I think it would at least be an improvement over what's happening now.)

So: back to search.

Obviously, there's a strong convention that a search box in a header or sidebar searches the entire site. Only searching titles and subtitles is quite surprising. So surprising that the failure to search the entire site looks more like a bug than a designer's decision to do something differently.

(I think that what you're calling "basic search" most people would call "filtering." Usually the UI for filtering is presented differently — e.g., a field over a list of values, etc.)

I can think of two options.

The first, and simplest, would be to hide the search box in templates unless Algolia is enabled. (I see that Algolia is already in the template feature dict, so that would be quite easy.)

The second would be to use a client-side search engine (e.g. Lunr.js) if Algolia isn't enabled. I'm not sure you really want to spend your time on that, but: it's an option. (Most databases also support full text search these days, but I assume you already decided not to go that route.)

My main concern is that prospects might try Typlog, assume there's a bug, and give up *because* they think there's an obvious bug that hasn't been fixed. That's worse, I think, than not having a search field.

Obviously, this is all opinion.

I did eventually figure out what was happening, and I'm able to make the template changes myself; feel free to close this unless there's something else worth talking about.

Thanks for taking the time to look at these!
